DataManager/src/DataManagement/VersionManager.java

```diff
@@ -1,16 +1,100 @@
 package DataManagement;
 
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.nio.file.AccessDeniedException;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * 
+ * @author Nathan Steer
+ *
+ */
 public class VersionManager
 {
-	private static int versionNum = 0;
+	private static Path path = Paths.get( "." + File.separator + "Files" + File.separator + "Version.dat" );
+	
 	
+	/**
+	 * Returns the servers current version number
+	 * <p>
+	 * Reads the version file stored on the server to find the version number,
+	 * and returns this number to the caller.
+	 * 
+	 * @return The version number of the servers language list in the form of an int
+	 */
 	public static int getVersion ()
 	{
-		return versionNum;
+		// If the file does not currently exist, create it
+		if ( Files.notExists( path ) )
+		{
+			updateVersion();
+		}
+		
+		// Attempt to read from the version file and return its contents
+		try ( InputStream in = Files.newInputStream( path ) )
+		{
+			return new InputStreamReader( in ).read();
+		}
+		catch ( NoSuchFileException e )
+		{
+			System.out.println( "File does not exist" );
+		}
+		catch ( AccessDeniedException e )
+		{
+			System.out.println( "Access to file was denied when attempting to read" );
+		}
+		catch ( IOException e )
+		{
+			System.out.println( "Failed to read file" );
+			e.printStackTrace();
+		}
+		
```

Comment:
I prefer the seperate catches here in order to provide meaningful error messages for the anticipated exceptions, since the built in messages on exceptions tend to be very vague and rarely helpful.